packages/components/src/higher-order/navigate-regions/index.js

```diff
@@ -54,7 +56,17 @@ export function useNavigateRegions( shortcuts = defaultShortcuts ) {
 			nextRegion = regions[ nextIndex ];
 		}
 
-		nextRegion.focus();
+		// Add the tabindex of -1 so tabbable elements can be located.
+		nextRegion.setAttribute( 'tabindex', '-1' );
+		// Find the next tabbable within the region.
+		const nextTabbable = focus.tabbable.findNext( nextRegion );
+		// Is there a next tabbable available? If not, focus the region instead.
+		const nextFocus = nextTabbable ? nextTabbable : nextRegion;
+		// Focus our chosen area, either the first tabbable within the region or the region itself.
+		nextFocus.focus();
+		// Remove the tabindex, no longer needed after focus.
+		nextRegion.removeAttribute( 'tabindex' );
```

Comment:
As mentioned in the comments, removing tabindex dynamically  breaks the feature on Chrome and screen readers don't announce anything.